scipy/spatial/transform/rotation.py

```diff
@@ -1,12 +1,71 @@
 from __future__ import division, print_function, absolute_import
 
 import numpy as np
+import scipy.linalg
+import warnings
 
 class Rotation(object):
     """Rotation in 3 dimensions.
 
     This class will include initializers from different representations, converters
     and some useful algorithms such as quaternion slerp and rotation estimation.
+
+    For initializing Rotations usage of `from_...` methods such as `from_quaternion`
+    is recommended instead of using `__init__`.
+
+    Methods
+    -------
+    from_quaternion
     """
-    def __init__(self):
-        pass
+    def __init__(self, quat, normalized=False):
+        self._single = False
+        # Try to convert to numpy array
+        quat = np.asarray(quat, dtype=float)
+
+        if quat.ndim not in [1, 2] or quat.shape[-1] != 4:
+            raise ValueError("`quat` should be of shape (4,) or (N x 4).")
+
+        # If a single quaternion is given, convert it to a 2D 1 x 4 matrix but
+        # set self._single to True so that we can return appropriate objects
+        # in the `to_...` methods
+        if quat.shape == (4,):
+            quat = quat[None, :]
+            self._single = True
+
+        if normalized:
+            self._quat = quat
+        else:
+            self._quat = quat.copy()
+            # L2 norm of each row
+            norms = scipy.linalg.norm(quat, axis=1)
+
+            # Warn user for zero (eps?) norm and set to identity quaternion,
+            # which (0,0,0,1) in (x,y,z,w) format
+            zero_norms = norms == 0
+            if zero_norms.any():
+                warnings.warn("Found zero norm quaternions in input, replacing with identity quaternions.")
+                self._quat[zero_norms] = np.array([0, 0, 0, 1])
```

Comment:
I'd rather just raise an error in this case.

Reply:
I thought about it too. I'm neutral (aka `+0`) to this and we have +1 to change from @larsoner.  

@adbugger --- what is your opinion?

Reply:
As I said, I think both options are reasonable. Perhaps I like throwing an error slightly more as well. So go ahead and change it to the error.